app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.js

```diff
@@ -534,14 +534,11 @@ module.exports = angular
 
       //update pipeline through a callback for React
       this.updatePipelineConfig = changes => {
-        $scope.$applyAsync(() => {
-          $scope.pipeline = _.cloneDeep($scope.pipeline);
```

Comment:
I took how we update stage configs as the Source of Truth for keeping React state in sync with Angular state, and there was no cloning there, just extending followed by marking dirty, which seems to be sufficient. That said, I don't know if changing the reference instead of marking dirty is preferable, I just figured it would be best if we had one way to do this and updating stages was the flow that was working :man_shrugging: https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/pipeline/config/stages/stage.module.js#L155

Reply:
ah ok, got it. I won't pretend I know enough about how it's all plumbed together to really catch issues in code review either way 😄

Reply:
well I appreciate you taking a look either way! just resolved conflicts against @louisjimenez's latest changes so also looping him in for a quick peek :pray: :pray: